application/src/main/java/org/opentripplanner/updater/spi/PollingGraphUpdater.java

```diff
@@ -113,11 +126,30 @@ public final void setup(WriteToGraphCallback writeToGraphCallback) {
    */
   protected abstract void runPolling() throws Exception;
 
-  protected final void updateGraph(GraphWriterRunnable task)
-    throws ExecutionException, InterruptedException {
-    var result = saveResultOnGraph.execute(task);
-    if (OTPFeature.WaitForGraphUpdateInPollingUpdaters.isOn()) {
-      result.get();
+  /**
+   * Post an update task to the GraphWriter queue.
+   * This is non-blocking.
+   * This can be called several times during one polling cycle.
+   */
+  protected final void updateGraph(GraphWriterRunnable task) {
```

Comment:
The system in this PR where this method is called from `runPolling()` in all subclasses, with this method being the sole way to submit tasks to a private `WriteToGraphCallback` seems like an even better approach than what I was speculating about in https://github.com/opentripplanner/OpenTripPlanner/pull/6262#issuecomment-2537979652.
I'd recommend we briefly explain that usage pattern in the Javadoc here though.

Reply:
Done